internal/hooks/v0hooks/manager.go

```diff
@@ -58,11 +58,33 @@ func configByName(
 		return &cfg.MFAVerificationAttempt, true
 	case PasswordVerification:
 		return &cfg.PasswordVerificationAttempt, true
+	case BeforeUserCreated:
+		return &cfg.BeforeUserCreated, true
+	case AfterUserCreated:
+		return &cfg.AfterUserCreated, true
 	default:
 		return nil, false
 	}
 }
 
+func (o *Manager) BeforeUserCreated(
+	ctx context.Context,
+	tx *storage.Connection,
+	req *BeforeUserCreatedInput,
+	res *BeforeUserCreatedOutput,
+) error {
+	return o.dispatch(ctx, &o.config.Hook.BeforeUserCreated, tx, req, res)
+}
+
+func (o *Manager) AfterUserCreated(
+	ctx context.Context,
+	tx *storage.Connection,
+	req *AfterUserCreatedInput,
+	res *AfterUserCreatedOutput,
+) error {
+	return o.dispatch(ctx, &o.config.Hook.AfterUserCreated, tx, req, res)
+}
+
```

Comment:
Why aren't these inside `InvokeHook`?

The tests can be changed to use `InvokeHook` instead with the correct inputs & outputs. `InvokeHook` can also make sure the `Metadata` struct field is filled in with the correct information.

Reply:
I would like to enforce strong typing when I begin calling these methods from the API package. I would like to remove the InvokeHook method in a future commit to remove the need for the type assertions.

Thoughts?

Reply:
Let's do that for all of the hooks in a future PR, but maybe start a Slack discussion on it first. How is it going to look like exactly? 

Right now the `input`/`output` are `any` but it's not like you can accidentally pass an unsupported value -- a programmer error either way. Maybe generics can be used?

Let's discuss on Slack

Reply:
Polymorphism can also be used by refactoring Input & Output into, say, `CustomAccessTokenHookInvocation` struct that implements an `Invoke()` interface that uses `ctx, tx, ...`

Reply:
I've updated PR to have them inside InvokeHook.